### PR TITLE
Tolerate throws in wasm/* tests, add 2gb-in-4gb variants.

### PR DESCRIPTION
**Pull request by kdashg**
For browsers that don't support this large of wasm memory sizes, consider as PASS, so that browser features don't block webgl conformance.
For UAs that throw on use of large TypedArrays, FAIL but safely.

Also, Firefox *does* support ~4GB wasm memory sizes, but currently throws when >2GB TypedArray views are passed to entrypoints. However, Firefox does support <2GB-sized views of >2GB ArrayBuffers, so add tests for that.
This reflects the ability for apps to use large WASM heaps, so long as only smaller views are passed to webgl.

### CODE REVIEW

**Comment by kdashg** [comment]
Chrome Canary: All tests pass
Firefox Nightly: 2gb-on-4gb pass, 4gb fail, 16gb "pass" (request for too many pages)

File: sdk/tests/conformance2/wasm/bufferdata-2gb-in-4gb-wasm-memory.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+<script src="../../js/webgl-test-utils.js"> </script>
+</head>
+<body>
+<canvas id="canvas" width="2" height="2" style="width: 40px; height: 40px;"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description(document.title);
+debug("Tests that bufferData can be called on a ~2GB view into ~4GB of WebAssembly Memory.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/wasm/bufferdata-2gb-in-4gb-wasm-memory.html:22]: This sounds ambiguous to me, like the view itself is 2 GB in length - could you consider a rephrase? Something like "called on a small view which is ~2 GB into a ~4 GB WebAssembly Memory"?

File: sdk/tests/conformance2/wasm/buffersubdata-2gb-in-4gb-wasm-memory.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+<script src="../../js/webgl-test-utils.js"> </script>
+</head>
+<body>
+<canvas id="canvas" width="2" height="2" style="width: 40px; height: 40px;"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description(document.title);
+debug("Tests that bufferSubData can be called on WebAssembly Memory of 4GB in size.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/wasm/buffersubdata-2gb-in-4gb-wasm-memory.html:22]: Can this be rephrased to be symmetric with the other tests'?

File: sdk/tests/conformance2/wasm/getbuffersubdata-2gb-in-4gb-wasm-memory.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+<script src="../../js/webgl-test-utils.js"> </script>
+</head>
+<body>
+<canvas id="canvas" width="2" height="2" style="width: 40px; height: 40px;"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description(document.title);
+debug("Tests that getBufferSubData can be called on WebAssembly Memory of 4GB in size.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/wasm/getbuffersubdata-2gb-in-4gb-wasm-memory.html:22]: Possible to update this comment to be more similar to the other tests', too?

File: sdk/tests/conformance2/wasm/readpixels-2gb-in-4gb-wasm-memory.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+<script src="../../js/webgl-test-utils.js"> </script>
+</head>
+<body>
+<canvas id="canvas" width="2" height="2" style="width: 40px; height: 40px;"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description(document.title);
+debug("Tests that gl.readPixels() can be called on WebAssembly Memory of 4GB in size.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/wasm/readpixels-2gb-in-4gb-wasm-memory.html:22]: Same here? Like "...called with a ~2 GB offset into a WebAssembly Memory 4 GB in size"?

File: sdk/tests/conformance2/wasm/teximage2d-2gb-in-4gb-wasm-memory.html [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+<script src="../../js/webgl-test-utils.js"> </script>
+</head>
+<body>
+<canvas id="canvas" width="2" height="2" style="width: 40px; height: 40px;"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description(document.title);
+debug("Tests that gl.texImage2D() can be called on WebAssembly Memory 4GB in size.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/wasm/teximage2d-2gb-in-4gb-wasm-memory.html:22]: Same here

File: sdk/tests/conformance2/wasm/texsubimage2d-2gb-in-4gb-wasm-memory.html [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+<script src="../../js/webgl-test-utils.js"> </script>
+</head>
+<body>
+<canvas id="canvas" width="2" height="2" style="width: 40px; height: 40px;"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description(document.title);
+debug("Tests that gl.texSubImage2D() can be called on WebAssembly Memory 4GB in size.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/wasm/texsubimage2d-2gb-in-4gb-wasm-memory.html:22]: Same request here

**Review by kenrussell — APPROVED** [review]
Looks great, thanks for revising the tests. A few small requests about making all the test descriptions more similar; please feel free to merge yourself. Thanks.

**Pull request merged**